pkg/manager/diff.go

```diff
@@ -512,14 +512,54 @@ func (rr *reproRunner) Run(r *repro.Result) {
 	rr.done <- ret
 }
 
-func PatchFocusAreas(cfg *mgrconfig.Config, gitPatch []byte) {
-	const maxAffectedByHeader = 50
+func PatchFocusAreas(cfg *mgrconfig.Config, gitPatches [][]byte) {
+	direct, transitive := affectedFiles(cfg, gitPatches)
+	if len(direct) > 0 {
+		sort.Strings(direct)
+		log.Logf(0, "adding directly modified files to focus_order: %q", direct)
+		cfg.Experimental.FocusAreas = append(cfg.Experimental.FocusAreas,
+			mgrconfig.FocusArea{
+				Name: "modified",
+				Filter: mgrconfig.CovFilterCfg{
+					Files: direct,
+				},
+				Weight: 3.0,
+			})
+	}
+
+	if len(transitive) > 0 {
+		sort.Strings(transitive)
+		log.Logf(0, "adding transitively affected to focus_order: %q", transitive)
+		cfg.Experimental.FocusAreas = append(cfg.Experimental.FocusAreas,
+			mgrconfig.FocusArea{
+				Name: "included",
+				Filter: mgrconfig.CovFilterCfg{
+					Files: transitive,
+				},
+				Weight: 2.0,
+			})
+	}
+
+	// Still fuzz the rest of the kernel.
+	if len(cfg.Experimental.FocusAreas) > 0 {
+		cfg.Experimental.FocusAreas = append(cfg.Experimental.FocusAreas,
+			mgrconfig.FocusArea{
+				Weight: 1.0,
+			})
+	}
+}
 
-	names := map[string]bool{}
-	includedNames := map[string]bool{}
-	for _, file := range vcs.ParseGitDiff(gitPatch) {
-		names[file] = true
+func affectedFiles(cfg *mgrconfig.Config, gitPatches [][]byte) (direct, transitive []string) {
+	const maxAffectedByHeader = 50
```

Comment:
Why 50?

Reply:
Just a random number to avoid the cases when the patch touches a header that's included in too many files -- then we shouldn't focus the fuzzer on all of them.